src/core/application.cpp

```diff
@@ -13,73 +14,186 @@ namespace app
 namespace core
 {
 
-void Application::configure()
+namespace fs = std::filesystem;
+using namespace std::literals;
+using namespace phosphor::logging;
+using namespace app::obmc::entity;
+
+constexpr const char* yawebInitGuradFile = YAWEB_INIT_GUARD_FILE;
+
+void Application::initDBusConnect()
 {
-    BMC_LOG_DEBUG << "configure";
+    dbusConnection = std::move(connect::DBusConnect::createDbusConnection());
+    createInitGuardFile();
+}
 
-    waitBootingBmc();
-    registerAllRoutes();
+void Application::runDBusObserve()
+{
+    dbusConnection->run();
+}
+
+void Application::createInitGuardFile()
+{
+    const auto filePath = fs::path(yawebInitGuradFile);
+
+    if (!fs::exists(filePath.parent_path()))
+    {
+        throw std::runtime_error(
+            "The path to create init-guard file is not exists. Filename: " +
+            filePath.string());
+    }
```

Comment:
"does not exist", by the way.

Reply:
> This is a check of the directory to create a file.

Wow, I've missed that you check the parent directory.
But the exception message is about the file and exactly this confused me.

Likely it would be more appropriate to make an attempt to create nonexistent directories and to throw the corresponding exception if fail.

Reply:
Fixed

Reply:
@nest1ing, I'd prefer to check if the path already exists, because the path specifies by meson-variable and let's get the system to care about that. And actually the path is /run/lighttpd/, so I believe the lighttpd should worry about own runtime directory.

Reply:
So, I can create a directory for reliability but this is actually not required, because as You are know the `yaweb` will starts only after `lighttpd` will be started. The yaweb.service.in unit file will worry about that.

Reply:
Not a big issue to hold this PR actually.